### PR TITLE
Add Hive view test for outer explode

### PR DESCRIPTION
**Pull request by alexjo2144**

### CODE REVIEW

File: testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestHiveViews.java [hunk 1]
```diff
@@ -152,6 +152,30 @@ public void testLateralViewExplode()
...
+        onPresto().executeQuery("DROP TABLE IF EXISTS pageAdsOuterExplode");
+        onPresto().executeQuery("CREATE TABLE pageAdsOuterExplode(pageid, adid_list) WITH (format='TEXTFILE') AS " +
+                "VALUES " +
+                "  (CAST('two' AS varchar), ARRAY[11, 22]), " +
+                "  ('nothing', NULL), " +
+                "  ('zero', ARRAY[]), " +
+                "  ('one', ARRAY[42])");
+
+        onHive().executeQuery("DROP VIEW IF EXISTS hive_lateral_view_outer_explode");
+        onHive().executeQuery("CREATE VIEW hive_lateral_view_outer_explode as SELECT pageid, adid FROM pageAdsOuterExplode LATERAL VIEW OUTER explode(adid_list) adTable AS adid");
```
Comment (phd3) [inline testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestHiveViews.java:167]: nit: spill to second line for better readability

**Review by phd3 — APPROVED** [review]
I'd just put it in `testLateralViewExplode` since both with/without `OUTER` keyword is supported by all [tested hive versions](https://cwiki.apache.org/confluence/display/Hive/LanguageManual+LateralView). not a strong opinion though, this seems fine too.

**Comment by alexjo2144** [comment]
Good point, no need to make the base table twice. Fixed, thanks.

**Comment by findepi** [comment]
```
2021-01-27T20:44:16.0734702Z tests               | 2021-01-28 02:29:16 INFO: FAILURE     /    io.trino.tests.hive.TestHiveViews.testLateralViewExplode (Groups: hive_views) took 2.6 seconds
2021-01-27T20:44:16.0790452Z tests               | 2021-01-28 02:29:16 SEVERE: Failure cause:
2021-01-27T20:44:16.0797588Z tests               | io.trino.tempto.query.QueryExecutionException: java.sql.SQLException: Error while compiling statement: FAILED: SemanticException [Error 10001]: Line 1:72 Table not found 'pageAdsOuterExplode'
2021-01-27T20:44:16.0800903Z tests               | 	at io.trino.tempto.query.JdbcQueryExecutor.execute(JdbcQueryExecutor.java:119)
2021-01-27T20:44:16.0804220Z tests               | 	at io.trino.tempto.query.JdbcQueryExecutor.executeQuery(JdbcQueryExecutor.java:84)
2021-01-27T20:44:16.0809808Z tests               | 	at io.trino.tests.utils.QueryExecutors$1.lambda$executeQuery$0(QueryExecutors.java:49)
2021-01-27T20:44:16.0811782Z tests               | 	at net.jodah.failsafe.Functions.lambda$get$0(Functions.java:48)
2021-01-27T20:44:16.0815040Z tests               | 	at net.jodah.failsafe.RetryPolicyExecutor.lambda$supply$0(RetryPolicyExecutor.java:62)
2021-01-27T20:44:16.0816785Z tests               | 	at net.jodah.failsafe.Execution.executeSync(Execution.java:129)
2021-01-27T20:44:16.0818042Z tests               | 	at net.jodah.failsafe.FailsafeExecutor.call(FailsafeExecutor.java:376)
2021-01-27T20:44:16.0819887Z tests               | 	at net.jodah.failsafe.FailsafeExecutor.get(FailsafeExecutor.java:67)
2021-01-27T20:44:16.0821612Z tests               | 	at io.trino.tests.utils.QueryExecutors$1.executeQuery(QueryExecutors.java:49)
2021-01-27T20:44:16.0823646Z tests               | 	at io.trino.tests.hive.TestHiveViews.testLateralViewExplode(TestHiveViews.java:155)
2021-01-27T20:44:16.0826399Z tests               | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2021-01-27T20:44:16.0828188Z tests               | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2021-01-27T20:44:16.0830446Z tests               | 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2021-01-27T20:44:16.0832251Z tests               | 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2021-01-27T20:44:16.0833929Z tests               | 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
2021-01-27T20:44:16.0835610Z tests               | 	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
2021-01-27T20:44:16.0836780Z tests               | 	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
2021-01-27T20:44:16.0838013Z tests               | 	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
2021-01-27T20:44:16.0839713Z tests               | 	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
2021-01-27T20:44:16.0841820Z tests               | 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
2021-01-27T20:44:16.0843428Z tests               | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
2021-01-27T20:44:16.0845256Z tests               | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
2021-01-27T20:44:16.0846521Z tests               | 	at java.base/java.lang.Thread.run(Thread.java:834)
2021-01-27T20:44:16.0847951Z tests               | Caused by: java.sql.SQLException: Error while compiling statement: FAILED: SemanticException [Error 10001]: Line 1:72 Table not found 'pageAdsOuterExplode'
2021-01-27T20:44:16.0849229Z tests               | 	at org.apache.hive.jdbc.Utils.verifySuccess(Utils.java:121)
2021-01-27T20:44:16.0850874Z tests               | 	at org.apache.hive.jdbc.Utils.verifySuccessWithInfo(Utils.java:109)
2021-01-27T20:44:16.0852292Z tests               | 	at org.apache.hive.jdbc.HiveStatement.execute(HiveStatement.java:231)
2021-01-27T20:44:16.0854357Z tests               | 	at io.trino.tempto.query.JdbcQueryExecutor.executeQueryNoParams(JdbcQueryExecutor.java:128)
2021-01-27T20:44:16.0856178Z tests               | 	at io.trino.tempto.query.JdbcQueryExecutor.execute(JdbcQueryExecutor.java:112)
2021-01-27T20:44:16.0857104Z tests               | 	... 22 more
2021-01-27T20:44:16.0858074Z tests               | 	Suppressed: java.lang.Exception: Query: CREATE VIEW hive_lateral_view_outer_explode as SELECT pageid, adid FROM pageAdsOuterExplode LATERAL VIEW OUTER explode(adid_list) adTable AS adid
2021-01-27T20:44:16.0859886Z tests               | 		at io.trino.tempto.query.JdbcQueryExecutor.executeQueryNoParams(JdbcQueryExecutor.java:136)
2021-01-27T20:44:16.0861572Z tests               | 		... 23 more
```

**Comment by alexjo2144** [comment]
Ah copy/pasted that test wrong. Fixed now, and build is green.

**Comment by phd3** [comment]
Merged, thanks!

**Pull request merged**